app/scripts/controllers/transactions.js

```diff
@@ -56,19 +67,22 @@ module.exports = class TransactionController extends EventEmitter {
         if (!account) return
         return account.balance
       },
-      publishTransaction: this.txProviderUtil.publishTransaction.bind(this.txProviderUtil),
-      getPendingTransactions: () => {
-        const network = this.getNetwork()
-        return this.getFilteredTxList({
-          status: 'submitted',
-          metamaskNetworkId: network,
-        })
+      publishTransaction: this.query.sendRawTransaction,
+      getPendingTransactions: this.txStateManager.getPendingTransactions.bind(this.txStateManager),
+      giveUpOnTransaction: (txId) => {
+        const msg = `Gave up submitting after 3500 blocks un-mined.`
+        this.setTxStatusFailed(txId, msg)
       },
     })
 
-    this.pendingTxTracker.on('txWarning', this.updateTx.bind(this))
-    this.pendingTxTracker.on('txFailed', this.setTxStatusFailed.bind(this))
-    this.pendingTxTracker.on('txConfirmed', this.setTxStatusConfirmed.bind(this))
+    this.txStateManager.subscribe(() => {
+      this.emit('update')
```

Comment:
dont need to emit update if our memStore is updated -- consumers should observe memStore

Reply:
✔️